cmake/RetrieveGoogleTest.cmake

```diff
@@ -0,0 +1,50 @@
+##
+## CMake script to download GoogleTest and
+## add to the project runtime.
+##
+## See http://kaizou.org/2014/11/gtest-cmake/
+
+# We need thread support
+find_package(Threads REQUIRED)
+
+# Enable ExternalProject CMake module
+include(ExternalProject)
+
+# Download and install GoogleTest
+ExternalProject_Add(
```

Comment:
Does this work offline too?

Linux packages usually make the download set separate from the build phase.

Reply:
Yea, valid point, unless you downloaded it prior...this would fail.